packages/react/src/components/DatePickerInput/DatePickerInput.js

```diff
@@ -13,6 +13,42 @@ import { Calendar16 } from '@carbon/icons-react';
 
 const { prefix } = settings;
 
+const upperSubLocale = (locale) => {
+  const splitLocale = locale.split(/[\s-_]+/);
+  if (splitLocale[1]) {
+    splitLocale[1] = splitLocale[1].toUpperCase();
+    return splitLocale.join('-');
+  }
+  return splitLocale[0];
+};
+
+const formatPlaceholder = (placeholder, locale) => {
+  if (placeholder !== undefined) return placeholder;
+  if (!locale) return null;
+  let formatObj;
+  try {
+    formatObj = new Intl.DateTimeFormat(upperSubLocale(locale)).formatToParts(
+      new Date()
+    );
+  } catch (err) {
+    console.log(err);
+    formatObj = new Intl.DateTimeFormat('en').formatToParts(new Date());
```

Comment:
This API is not supported by IE11. As @emyarod said, my suggestion was that dealing with date format in application code instead of doing it in our date picker code.

Reply:
I've got it now to fall back to whats currently in carbon, so hopefully this can only be an improvement when possible. If you see any issue with falling back to default props then please let me know

Reply:
the try-catch block would cause IE11 to fall back to the original/current component behavior, and for the browsers that do support the `formatToParts` method, the formatting would be displayed differently. not sure if we want to proceed if this would create inconsistent behavior across browsers

Reply:
While I lean towards ignoring IE11 since it's EOL ~ Aug '21, the other option would be to create a date with Jan 31, 1970 or something, then parse the location of each since `Intl.DateTimeFormat(locale).format(new Date())` is supported on all browsers

Example: 
![image](https://user-images.githubusercontent.com/30727984/96774012-62a53500-13b3-11eb-977e-23bd085926ab.png)

Reply:
since this would be setting the default formatting based on locale, I think unit tests against all of the [available locales](https://github.com/unicode-cldr/cldr-core/blob/master/availableLocales.json) should be expected. but aside from that, Carbon has a commitment to supporting IE11 so I don't think we should ignore it until support for IE11 is officially dropped from the library